engine/rpcserver.go

```diff
@@ -3077,3 +3044,36 @@ func (s *RPCServer) GetRecentTrades(_ context.Context, r *gctrpc.GetSavedTradesR
 
 	return resp, nil
 }
+
+func (s *RPCServer) checkParams(exch string, a asset.Item, p currency.Pair) (exchange.IBotExchange, error) {
```

Comment:
I would have this function take in an IbotExchange instead of returning one.  And not make this a method of RPCServer.

Reply:
I agree. With a name `checkParams`, it should only be checking, not creating new params to return